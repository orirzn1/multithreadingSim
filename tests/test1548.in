L15
S6
O13
N12

T0
I@0x00000000
SUB $6, $2, $6
LOAD $7, $7, $2
LOAD $2, $6, 0x0
STORE $2, $2, $2
ADD $3, $4, $5
HALT $0

T1
I@0x00000000
STORE $3, $7, $0
SUB $7, $7, $0
STORE $4, $3, 0x0
SUB $3, $2, $0
HALT $0

T2
I@0x00000000
SUB $1, $0, $7
ADDI $7, $1, 0x2
SUB $6, $3, $1
LOAD $1, $6, 0x4
STORE $4, $0, 0x2
STORE $4, $5, $5
LOAD $5, $1, $7
ADD $2, $6, $6
SUB $3, $4, $6
SUB $3, $0, $5
HALT $0

T3
I@0x00000000
SUBI $2, $2, 0x4
ADD $5, $2, $0
LOAD $5, $3, $1
ADDI $2, $7, 0x4
ADDI $7, $7, 0x3
HALT $0

T4
I@0x00000000
LOAD $3, $1, 0x2
SUBI $0, $0, 0x2
ADDI $0, $0, 0x1
ADDI $0, $2, 0x4
LOAD $5, $1, 0x0
STORE $1, $3, $2
SUB $5, $6, $6
LOAD $1, $0, 0x4
SUBI $5, $2, 0x4
SUB $2, $4, $3
HALT $0

T5
I@0x00000000
ADDI $7, $5, 0x2
LOAD $3, $6, $5
LOAD $0, $1, 0x4
STORE $4, $0, $5
HALT $0

T6
I@0x00000000
STORE $3, $1, 0x2
ADD $0, $2, $0
ADDI $4, $7, 0x1
ADDI $6, $0, 0x2
HALT $0

T7
I@0x00000000
SUBI $0, $1, 0x3
SUB $3, $7, $4
ADDI $2, $1, 0x4
LOAD $0, $7, $0
ADD $1, $5, $5
ADD $4, $2, $5
LOAD $6, $4, $2
HALT $0

T8
I@0x00000000
LOAD $0, $7, $3
STORE $4, $5, 0x1
STORE $1, $3, $3
LOAD $3, $0, 0x0
LOAD $5, $6, 0x3
LOAD $4, $2, $0
ADDI $1, $5, 0x1
SUB $1, $1, $1
ADDI $6, $6, 0x2
HALT $0

T9
I@0x00000000
LOAD $3, $1, 0x4
ADDI $4, $5, 0x2
SUBI $6, $2, 0x2
LOAD $1, $7, 0x4
LOAD $5, $4, 0x3
ADDI $5, $3, 0x2
ADDI $1, $5, 0x2
LOAD $6, $4, 0x4
STORE $1, $0, $2
HALT $0

T10
I@0x00000000
SUBI $2, $5, 0x2
SUB $7, $2, $2
LOAD $2, $4, $6
SUB $4, $7, $3
SUBI $4, $6, 0x1
ADD $6, $6, $1
STORE $1, $7, 0x0
ADDI $5, $7, 0x1
HALT $0

T11
I@0x00000000
LOAD $5, $2, 0x3
SUBI $5, $4, 0x2
LOAD $4, $7, $7
SUB $4, $4, $7
STORE $0, $6, $4
ADD $7, $5, $0
SUB $2, $7, $2
LOAD $3, $2, 0x1
SUB $5, $5, $2
STORE $4, $6, $5
HALT $0

D@0x0
0x2
0x6
0xA
0x3
0x8
0xA
0x3
0x4
0x4
