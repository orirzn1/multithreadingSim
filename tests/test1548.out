
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x6	R2 = 0x0	R3 = 0xFFFFFFF6	R4 = 0x0	R5 = 0xA	R6 = 0x0	R7 = 0x2
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 4:
	R0 = 0x4	R1 = 0xA	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x6	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x4	R5 = 0x0	R6 = 0xA	R7 = 0x0
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0xA	R2 = 0x0	R3 = 0x6	R4 = 0x2	R5 = 0x8	R6 = 0x6	R7 = 0x0
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFF	R5 = 0x1	R6 = 0x0	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Blocked MT CPI for this program 7.567010

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x6	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 4:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x6	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x4	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0xA	R2 = 0x0	R3 = 0x6	R4 = 0x2	R5 = 0x8	R6 = 0x6	R7 = 0x0
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFF	R5 = 0x1	R6 = 0x0	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Finegrained Multithreading CPI for this program 1.113402

